components/aws/sagemaker/tests/unit_tests/tests/test_train.py

```diff
@@ -27,6 +27,272 @@ def setUpClass(cls):
+    ]
+
+    with self.assertRaises(Exception):
+      _utils.wait_for_training_job(mock_client, 'training-job', 0)
+
+    self.assertEqual(mock_client.describe_training_job.call_count, 4)
+
+  def test_get_model_artifacts_from_job(self):
+    mock_client = MagicMock()
+    mock_client.describe_training_job.return_value = {"ModelArtifacts": {"S3ModelArtifacts": "s3://path/"}}
+
+    self.assertEqual(_utils.get_model_artifacts_from_job(mock_client, 'training-job'), 's3://path/')
+
+  def test_get_image_from_defined_job(self):
+    mock_client = MagicMock()
+    mock_client.describe_training_job.return_value = {"AlgorithmSpecification": {"TrainingImage": "training-image-url"}}
+
+    self.assertEqual(_utils.get_image_from_job(mock_client, 'training-job'), "training-image-url")
+
+  def test_get_image_from_algorithm_job(self):
+    mock_client = MagicMock()
+    mock_client.describe_training_job.return_value = {"AlgorithmSpecification": {"AlgorithmName": "my-algorithm"}}
+    mock_client.describe_algorithm.return_value = {"TrainingSpecification": {"TrainingImage": "training-image-url"}}
+
+    self.assertEqual(_utils.get_image_from_job(mock_client, 'training-job'), "training-image-url")
+
+  def test_reasonable_required_args(self):
+    response = _utils.create_training_job_request(vars(self.parser.parse_args(required_args)))
+
+    # Ensure all of the optional arguments have reasonable default values
+    self.assertFalse(response['EnableManagedSpotTraining'])
+    self.assertDictEqual(response['HyperParameters'], {})
+    self.assertNotIn('VpcConfig', response)
+    self.assertNotIn('MetricDefinitions', response)
+    self.assertEqual(response['Tags'], [])
+    self.assertEqual(response['AlgorithmSpecification']['TrainingInputMode'], 'File')
+    self.assertEqual(response['OutputDataConfig']['S3OutputPath'], 'test-path')
+
+  def test_metric_definitions(self):
+    metric_definition_args = self.parser.parse_args(required_args + ['--metric_definitions', '{"metric1": "regexval1", "metric2": "regexval2"}'])
+    response = _utils.create_training_job_request(vars(metric_definition_args))
+
+    self.assertIn('MetricDefinitions', response['AlgorithmSpecification'])
+    response_metric_definitions = response['AlgorithmSpecification']['MetricDefinitions']
+
+    self.assertEqual(response_metric_definitions, [{
+      'Name': "metric1",
+      'Regex': "regexval1"
+    }, {
+      'Name': "metric2",
+      'Regex': "regexval2"
+    }])
+
+  def test_no_defined_image(self):
+    # Pass the image to pass the parser
+    no_image_args = required_args.copy()
+    image_index = no_image_args.index('--image')
+    # Cut out --image and it's associated value
+    no_image_args = no_image_args[:image_index] + no_image_args[image_index+2:]
+
+    parsed_args = self.parser.parse_args(no_image_args)
+
+    with self.assertRaises(Exception):
+      _utils.create_training_job_request(vars(parsed_args))
+
+  def test_multiple_defined_images(self):
```

Comment:
I didn't understand this test ? SageMaker API suggest only one image can be passed in create_training_job api.

Reply:
I have renamed this test to accurately describe the contents.